src/main/java/cheese/data/TaskList.java

```diff
@@ -0,0 +1,151 @@
+        taskList.add(task);
+        return task;
+    }
+
+    /**
+     * Marks task in the list at given index as complete.
+     * 
+     * @param taskIndex Index of task to mark as complete
+     * @return Task that is marked as complete.
+     * @throws CheeseException If index is not in the range of task list.
+     */
+    public Task markTaskAsDone(int taskIndex) throws CheeseException {
+        Task task = getTask(taskIndex);
+        task.setDone();
+        return task;
+    }
+
+    /**
+     * Marks task in the list at given index as incomplete.
+     * 
+     * @param taskIndex Index of task to mark as incomplete
+     * @return Task that is marked as incomplete.
+     * @throws CheeseException If index is not in the range of task list.
+     */
+    public Task markTaskAsNotDone(int taskIndex) throws CheeseException {
+        Task task = getTask(taskIndex);
+        task.setNotDone();
+        return task;
+    }
+
+    /**
+     * Gets size of task list.
+     * 
+     * @return Size of task list.
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Returns list of tasks containing given keyword.
+     * 
+     * @param keyword Keyword to search for.
+     * @return List of tasks containing given keyword.
+     */
+    public ArrayList<Task> searchTasksByKeyword(String keyword) {
+        ArrayList<Task> searchResult = new ArrayList<>();
+        for (Task task : taskList) {
+            if (task.hasKeyword(keyword)) {
+                searchResult.add(task);
+            }
+        }
+        return searchResult;
+    }
+
+    /**
+     * Returns String representation of task list to save in file.
+     * 
+     * @return String representation of task list to save in file.
+     */
+    public String toFileString() {
+        String fileString = "";
+        for (Task task : taskList) {
+            fileString += task.toFileString();
+            fileString += System.lineSeparator();
+        }
+        return fileString;
+    }
+
+    /**
+     * Returns String representation of task list.
+     * 
+     * @return String representation of task list.
+     */
+    @Override
+    public String toString() {
+        String taskListString = "";
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            int displayIndex = i + 1;
+            taskListString += displayIndex + ". " + task;
+            taskListString += System.lineSeparator();
+        }
+        return taskListString.trim();
+    }
+
+    /**
+     * Validates if index given is within range of task list.
+     * 
+     * @param taskIndex Index of task to validate.
+     * @throws CheeseException If index is not in the range of task list.
+     */
+    private void validateTaskIndexInRange(int taskIndex) throws CheeseException {
+        if (taskIndex < 0 || taskIndex >= taskList.size()) {
+            throw new CheeseException("Item number is not in list range.");
+        }
```

Comment:
Really like how consistent your naming conventions are (i.e. `validate` for functions that throw errors), makes the code really easy to understand!

Reply:
thanks for reviewing my code!